crates/api_crud/src/community/update.rs

```diff
@@ -48,6 +51,21 @@ impl PerformCrud for EditCommunity {
     }
 
     let community_id = data.community_id;
+    if let Some(languages) = data.discussion_languages.clone() {
+      let site_languages: Vec<LanguageId> =
+        blocking(context.pool(), SiteLanguage::read_local).await??;
+      // check that community languages are a subset of site languages
+      // https://stackoverflow.com/a/64227550
+      let is_subset = languages.iter().all(|item| site_languages.contains(item));
+      if !is_subset {
+        return Err(LemmyError::from_message("language_not_allowed"));
+      }
+      blocking(context.pool(), move |conn| {
+        CommunityLanguage::update(conn, languages, community_id)
```

Comment:
You can delete the subset check above, and just make sure CommunityLanguage::update is doing the subset in SQL by reading from SiteLanguages.

Reply:
You mean as part of the insert statement? How would that work in sql?

Reply:
No, I mean inside the `CommunityLanguage::update(` fn, where it looks like you are already reading the site languages anyway. Looks like since users are providing any list of languages, you do have to read the site languages from sql, but can do a `.retain` type thing in that function.

Reply:
I see, but i cant move the check in here because of LemmyError in return value. Then i would also have to return LemmyError from Community::create(), which is incompatible with the trait.